src/containers/base-class/index.js

```diff
@@ -0,0 +1,185 @@
+import { h, Component } from "preact";
+
+import InvoiceService from "../../services/invoice";
+
+import DonateScreen from "../../views/donate-screen";
+import LoadingScreen from "../../views/loading-screen";
+import WelcomeScreen from "../../views/welcome-screen";
+import PaymentScreen from "../../views/payment-screen";
+import ThankyouScreen from "../../views/thankyou-screen";
+
+import { DEFAULT_API_BASE_URL } from "../../utils/constants";
+
+function getInitialState() {
+  const initialState = Object.assign({}, WidgetBase.defaultProps, ...arguments);
+
+  return {
+    ...initialState,
+    currentScreen: initialState.welcomeMessage
+      ? "welcome-screen"
+      : "donate-screen",
+  };
+}
+
+class WidgetBase extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = getInitialState(props);
+    this.invoiceService = new InvoiceService({
+      apiToken: this.props.apiToken,
+      apiBaseUrl: this.props.apiBaseUrl,
+      paymentRequestRenderer: this.paymentRequestRenderer,
+    });
+  }
+
+  componentDidMount() {
+    return fetch(`${this.props.apiBaseUrl}/v1/configuration`, {
+      headers: { "Api-Token": this.props.apiToken },
+    }).then((response) => {
+      if (response.ok) {
+        response.json().then((config) => {
+          this.widgetConfig = config;
+          this.setDefaults();
+        });
+      } else {
+        this.setDefaults();
+      }
+    });
+  }
+
+  setDefaults = () => {
+    this.setState(getInitialState(this.props, this.widgetConfig));
+  };
+
+  closeModal = () => {
+    this.invoiceService.reset();
+    this.setDefaults();
+  };
+
+  openModal = () => {
+    this.setState({
+      isModalOpen: true,
+    });
+  };
+
+  requestPayment = (args) => {
+    const amountInCents = parseInt(args.amount, 10);
+
+    if (!amountInCents) {
+      console.log("Invalid amount");
+    } else {
+      this.setState(
+        {
+          isModalOpen: true,
+          currentScreen: "loading-screen",
+        },
+        () => {
+          this.requestPaymentHandler(amountInCents);
+        }
+      );
+    }
+  };
+
+  openWidget = (attributes) => {
```

Comment:
I think those should go into the Widget class. they are specific for the modal widget.